mongodb/queue.go

```diff
@@ -315,6 +327,7 @@ func (q *queueMongoDB) waitForMessage() (bool, error) {
 		if !job.ResultMessage.Done {
 			q.moveToResult(&job, nil, monsterqueue.ErrNoJobResultSet)
 		}
+		q.taskDone <- true
```

Comment:
I think you need to wrap this line in a `if q.config.WaitTask` otherwise this would block forever. I think the CI failure is due to this.